subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/BuildCacheLoadMetadata.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.controller;
+
+import com.google.common.collect.ImmutableSortedMap;
+import org.gradle.caching.internal.origin.OriginMetadata;
+import org.gradle.caching.internal.packaging.BuildCacheEntryPacker;
+import org.gradle.internal.snapshot.FileSystemSnapshot;
+
+public class BuildCacheLoadMetadata {
```

Comment:
I'd call this `LoadResult` as a nested type in `BuildCacheController`.

Reply:
I tried to have it as a nested type, but the classcycles did not like it since it's also referenced in `.*.controler.services` package. Ideally, we would have one `BuildCacheServiceHandle` interface and we can have it there as nested. But for now I created an interfaces in `.*controller.service` package.